src/test/ui/lifetime-errors/ex1-return-one-existing-name-return-type-is-anon.nll.stderr

```diff
@@ -4,17 +4,16 @@ warning: not reporting region error due to nll
 LL |     x //~ ERROR lifetime mismatch
    |     ^
 
-error[E0623]: lifetime mismatch
+error: unsatisfied lifetime constraints
   --> $DIR/ex1-return-one-existing-name-return-type-is-anon.rs:18:5
    |
 LL |   fn foo<'a>(&self, x: &'a i32) -> &i32 {
-   |                        -------     ----
-   |                        |
-   |                        this parameter and the return type are declared with different lifetimes...
+   |          --  - let's call the lifetime of this reference `'1`
+   |          |
+   |          lifetime `'a` defined here
 LL | 
 LL |     x //~ ERROR lifetime mismatch
-   |     ^ ...but data from `x` is returned here
+   |     ^ function was supposed to return data with lifetime `'a` but it is returning data with lifetime `'1`
```

Comment:
maybe we should file a bug here -- I think it'd be nice to add another note here and use this as an opportunity to teach the elision rules. e.g.,

```
error: unsatisfied lifetime constraints
  --> $DIR/ex1-return-one-existing-name-return-type-is-anon.rs:18:5
   |
LL |   fn foo<'a>(&self, x: &'a i32) -> &i32 {
   |          --  -                     - defaults to `&'1 i32` per the elision rules
   |          |  |
   |          |  let's call the lifetime of this reference `'1`
   |          lifetime `'a` defined here
LL | 
LL |     x //~ ERROR lifetime mismatch
| ^ function was supposed to return data with lifetime `'a` but it is returning data with lifetime `'1`
```